src/app/guard/tests/exclusive-online-game-guard.ts

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable max-lines-per-function */
+import { ConnectedUserService, AuthUser } from 'src/app/services/ConnectedUserService';
+import { Router } from '@angular/router';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { BlankComponent } from 'src/app/utils/tests/TestUtils.spec';
+import { ConnectedUserServiceMock } from 'src/app/services/tests/ConnectedUserService.spec';
+import { ExclusiveOnlineGameGuard } from '../verified-and-not-active.guard';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { PartDocument } from 'src/app/domain/Part';
+import { PartMocks } from 'src/app/domain/PartMocks.spec';
+import { MessageDisplayer } from 'src/app/services/MessageDisplayer';
+import { UserMocks } from 'src/app/domain/UserMocks.spec';
+import { VerifiedAccountGuard } from '../verified-account.guard';
+
+describe('ExclusiveOnlineGameGuard', () => {
+
+    let guard: ExclusiveOnlineGameGuard;
+
+    let connectedUserService: ConnectedUserService;
+
+    let router: Router;
+
+    const startedPartUserPlay: PartDocument = new PartDocument('I-play', PartMocks.STARTED);
+
+    beforeEach(fakeAsync(async() => {
+        await TestBed.configureTestingModule({
+            imports: [
+                RouterTestingModule.withRoutes([
+                    { path: '**', component: BlankComponent },
+                ]),
+            ],
+            providers: [
+                { provide: ConnectedUserService, useClass: ConnectedUserServiceMock },
+            ],
+        }).compileComponents();
+        router = TestBed.inject(Router);
+        spyOn(router, 'navigate');
+        connectedUserService = TestBed.inject(ConnectedUserService);
+        const messageDisplayer: MessageDisplayer = TestBed.inject(MessageDisplayer);
+        guard = new ExclusiveOnlineGameGuard(connectedUserService, messageDisplayer, router);
+        ConnectedUserServiceMock.setUser(UserMocks.CONNECTED_AUTH_USER);
+    }));
+    it('should create', () => {
+        expect(guard).toBeDefined();
+    });
+    it('should delegate the first part of evaluate to mother class', fakeAsync(async() => {
+        ConnectedUserServiceMock.setUser(AuthUser.NOT_CONNECTED);
+        spyOn(VerifiedAccountGuard, 'evaluateUserPermission').and.callThrough();
+        await guard.canActivate();
+        expect(VerifiedAccountGuard.evaluateUserPermission).toHaveBeenCalledOnceWith(router, AuthUser.NOT_CONNECTED);
+    }));
+    it('should refuse to go to creation component when you have any observed part and redirect to it', fakeAsync(async() => {
```

Comment:
should forbid

Reply:
45 should refuse
257 should forbid.
Celui que tu mentionne est changé, tu veux qu'j'en change 45 autres :p ?

Reply:
non ça va !